src/shared/callbacks.rs

```diff
@@ -96,39 +96,71 @@ impl Callback {
             Err(_e) => Err("Callback failed".to_string()),
         }
     }
+
+    pub fn call1(&self, ss: String) -> Result<Promise, String> {
+        match self.f.call1(&JsValue::NULL, &JsValue::from_str(&ss)) {
+            Ok(result) => Ok(Promise::from(result)),
+            Err(_e) => Err("Callback failed".to_string()),
+        }
+    }
 }
 
 unsafe impl<'a> Send for Callback {}
 unsafe impl<'a> Sync for Callback {}
 
+pub enum CallbackType {
+    Buckets,
+    Measurements { bucket: String },
+}
+
 #[derive(Default, Clone)]
 pub struct Callbacks {
     pub buckets: Option<Callback>,
+    pub measurements: Option<Callback>,
 }
 
 impl Callbacks {
     pub fn register_buckets_callback(&mut self, f: Function) {
         self.buckets = Some(Callback::new(f));
     }
 
-    fn call_buckets(&self) -> Result<JsFuture, String> {
-        if let Some(cb) = self.buckets.clone() {
-            let promise = cb.call0()?;
-            Ok(JsFuture::from(promise))
-        } else {
-            Err("No buckets function set".to_string())
+    pub fn register_measurements_callback(&mut self, f: Function) {
+        self.measurements = Some(Callback::new(f));
+    }
+
+    fn call_js(&self, typ: CallbackType) -> Result<JsFuture, String> {
```

Comment:
Similar thing here, why not have call_buckets and call_measurements? call measurements could accept an argument for bucket name.

Reply:
it's an internal func, just for `cln.call_js(typ)`, so we won't write duplicate codes for `get(&self, typ: CallbackType)`.